tests/search/apps/test_AppResultItem.py

```diff
@@ -1,21 +1,41 @@
+import pathlib
 import mock
 import pytest
+import gi
+gi.require_version('Gio', '2.0')
+gi.require_version('GdkPixbuf', '2.0')
+# pylint: disable=wrong-import-position
+from gi.repository import Gio, GdkPixbuf
+from ulauncher.utils.db.KeyValueDb import KeyValueDb
 from ulauncher.search.apps.AppResultItem import AppResultItem
 from ulauncher.search.QueryHistoryDb import QueryHistoryDb
-from ulauncher.search.apps.AppStatDb import AppStatDb
 from ulauncher.search.Query import Query
 
+# Note: These mock apps actually need real values for Exec or Icon, or they won't load,
+# and they need to load from actual files or get_id() and get_filename() will return None
+ENTRIES_DIR = pathlib.Path(__file__).parent.joinpath('mock_desktop_entries').resolve()
+
 
 class TestAppResultItem:
+    @pytest.fixture(autouse=True)
+    def patch_DesktopAppInfo_new(self, mocker):
+        def mkappinfo(app_id):
+            return Gio.DesktopAppInfo.new_from_filename(f'{ENTRIES_DIR}/{app_id}')
+        return mocker.patch("ulauncher.search.apps.AppResultItem.Gio.DesktopAppInfo.new", new=mkappinfo)
+
+    @pytest.fixture(autouse=True)
+    def patch_DesktopAppInfo_get_all(self, mocker):
+        def get_all_appinfo():
+            return map(Gio.DesktopAppInfo.new, ['trueapp.desktop', 'falseapp.desktop'])
```

Comment:
I patched `Gio.DesktopAppInfo.new` above. That was `ulauncher.search.apps.AppResultItem.Gio.DesktopAppInfo.new`. However patching a class method just patches it all places the module is used, and the object path you use is just a way to access it? When I tested this it worked andyway, and it was shorter than using `AppResultItem.from_id(...)._app_info`